### ISSUE TITLE
MACROS: Reading from measurement fails while parsing response

### ISSUE THREAD

**Issue by ioreskovic** [fsanaulla/chronicler#104]
It seems Influx always returns timestamp as first field, while the rest of them are both tags and fields sorted alphabetically

Influx 1.6 returns the following order or field values:
1. time
2. bar
3. foo
4. tagKey1
5. tagKey2

as it can be seen when you execute the generated URI yourself:

```bash
http http://localhost:8086/query\?epoch\=ns\&u\=admin\&pretty\=true\&db\=chroniclerTests\&q\=SELECT+\*+FROM+domainMeasurement\&chunked\=false\&p\=admin | clip
```

```json
{
    "results": [
        {
            "statement_id": 0,
            "series": [
                {
                    "name": "domainMeasurement",
                    "columns": [
                        "time",
                        "aTag",
                        "cField",
                        "dField",
                        "dTag"
                    ],
                    "values": [
                        [
                            1541361859851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361860851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361861851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361862851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361863851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361864851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361865851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361866851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361867851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ],
                        [
                            1541361868851000000,
                            "firstTag",
                            42,
                            13,
                            "secondTag"
                        ]
                    ]
                }
            ]
        }
    ]
}
```

Using InfluxDB 1.6 running in Docker and chronicler forked from https://github.com/fsanaulla/chronicler/commit/df0e23ca7b76f0746169bea13f3bb1530f6ce2fe

```yml
version: '3'

services:
  # Define a Telegraf service
  telegraf:
    image: telegraf:latest
    volumes:
      - ./etc/telegraf.conf:/etc/telegraf/telegraf.conf:ro
    links:
      - influxdb
    ports:
      - "8092:8092/udp"
      - "8094:8094"
      - "8125:8125/udp"
  # Define an InfluxDB service
  influxdb:
    image: influxdb:latest
    volumes:
      - ./data/influxdb:/var/lib/influxdb
    ports:
      - "8086:8086"
  # Define a Chronograf service
  chronograf:
    image: chronograf:latest
    environment:
      INFLUXDB_URL: http://influxdb:8086
      KAPACITOR_URL: http://kapacitor:9092
    ports:
      - "8888:8888"
    links:
      - influxdb
      - kapacitor
  # Define a Kapacitor service
  kapacitor:
    image: kapacitor:latest
    environment:
      KAPACITOR_HOSTNAME: kapacitor
      KAPACITOR_INFLUXDB_0_URLS_0: http://influxdb:8086
    links:
      - influxdb
    ports:
      - "9092:9092"
  # Define a service for using the influx CLI tool.
  # docker-compose run influxdb-cli
  influxdb-cli:
    image: influxdb:latest
    entrypoint:
      - influx
      - -host
      - influxdb
    links:
      - influxdb
  # Define a service for using the kapacitor CLI tool.
  # docker-compose run kapacitor-cli
  kapacitor-cli:
    image: kapacitor:latest
    entrypoint: bash
    environment:
      KAPACITOR_URL: http://kapacitor:9092
    links:
      - kapacitor
```

Running this code for MCVE:

```scala
package com.github.fsanaulla.chronicler.examples.read

import java.time.Instant

import com.github.fsanaulla.chronicler.core.enums.Epochs
import com.github.fsanaulla.chronicler.core.model.{InfluxConfig, InfluxCredentials, InfluxFormatter, InfluxReader}
import com.github.fsanaulla.chronicler.macros.annotations.{field, tag, timestamp}
import com.github.fsanaulla.chronicler.urlhttp.io.InfluxIO
import com.github.fsanaulla.chronicler.urlhttp.management.InfluxMng
import jawn.ast.JArray

case class DomainEntity(
    @tag aTag: String,
    @tag dTag: String,
    @field cField: Int,
    @field dField: Double,
    @timestamp epochNanos: Long
)

class MyDomainReader extends InfluxReader[DomainEntity] {
  override def read(js: JArray): DomainEntity = {
    println(js)
    DomainEntity("a", "b", 1, 2.3, 11L)
  }
}

object DomainRead extends App {
  val influxConfig = InfluxConfig(
    "localhost",
    8086,
    Some(InfluxCredentials(username = "admin", password = "admin")),
    false)

  val influxIO = InfluxIO(influxConfig)
  val influxMng = InfluxMng(influxConfig)

  val databaseName = "chroniclerTests"
  val measurementName = "domainMeasurement"

  def generateEntity(time: Instant): DomainEntity = DomainEntity(
    "firstTag",
    "secondTag",
    42,
    13.0,
    time.toEpochMilli * 1000000
  )

  def generateEntities(time: Instant)(n: Int): Seq[DomainEntity] =
    (0 until n).map(s => generateEntity(time.plusSeconds(s)))

  val mm = influxIO.measurement[DomainEntity](
    dbName = "chroniclerTests",
    measurementName = "domainMeasurement"
  )

  import com.github.fsanaulla.chronicler.macros.Macros._
  implicit val domainFormatter: InfluxFormatter[DomainEntity] = format[DomainEntity]

  private val query = s"SELECT * FROM $measurementName"

  val p = for {
    _ <- influxMng.dropMeasurement(databaseName, measurementName)
    wr <- mm.bulkWrite(generateEntities(Instant.now())(10))
    uri = mm.readFromInfluxSingleQuery(databaseName, query, Epochs.NANOSECONDS, true, false)(influxConfig.credentials)
    js <- mm.readJs(databaseName, query, Epochs.NANOSECONDS, true, false).map(rr => rr.queryResult.mkString(", "))
  } yield (wr, uri, js)

  println(p)

  val domainResultFake = mm.read(query)(new MyDomainReader).fold(
    err => {err.printStackTrace(System.out); err.getMessage},
    rr => rr.queryResult.mkString(System.lineSeparator())
  )

  val domainResultMacros = mm.read(query)(domainFormatter).fold(
    err => {err.printStackTrace(System.out); err.getMessage},
    rr => rr.queryResult.mkString(System.lineSeparator())
  )

  println
  println(domainResultFake)

  println
  println(domainResultMacros)
}

```

Here's the output
```log
[info] Running com.github.fsanaulla.chronicler.examples.read.DomainRead
Success((WriteResult(204,true,None),http://localhost:8086/query?epoch=ns&u=admin&pretty=true&db=chroniclerTests&q=SELECT+*+FROM+domainMeasurement&chunked=false&p=admin,[1541361859851000000,"firstTag",42,13,"secondTag"], [1541361860851000000,"firstTag",42,13,"secondTag"], [1541361861851000000,"firstTag",42,13,"secondTag"], [1541361862851000000,"firstTag",42,13,"secondTag"], [1541361863851000000,"firstTag",42,13,"secondTag"], [1541361864851000000,"firstTag",42,13,"secondTag"], [1541361865851000000,"firstTag",42,13,"secondTag"], [1541361866851000000,"firstTag",42,13,"secondTag"], [1541361867851000000,"firstTag",42,13,"secondTag"], [1541361868851000000,"firstTag",42,13,"secondTag"]))
[1541361859851000000,"firstTag",42,13,"secondTag"]
[1541361860851000000,"firstTag",42,13,"secondTag"]
[1541361861851000000,"firstTag",42,13,"secondTag"]
[1541361862851000000,"firstTag",42,13,"secondTag"]
[1541361863851000000,"firstTag",42,13,"secondTag"]
[1541361864851000000,"firstTag",42,13,"secondTag"]
[1541361865851000000,"firstTag",42,13,"secondTag"]
[1541361866851000000,"firstTag",42,13,"secondTag"]
[1541361867851000000,"firstTag",42,13,"secondTag"]
[1541361868851000000,"firstTag",42,13,"secondTag"]
jawn.ast.WrongValueException: expected string, got number
        at jawn.ast.JValue.asString(JValue.scala:25)
        at com.github.fsanaulla.chronicler.examples.read.DomainRead$$anon$1.read(DomainRead.scala:57)
        at com.github.fsanaulla.chronicler.examples.read.DomainRead$$anon$1.read(DomainRead.scala:57)
        at com.github.fsanaulla.chronicler.urlhttp.io.api.Measurement.$anonfun$read$2(Measurement.scala:74)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
        at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
        at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
        at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:194)
        at scala.collection.TraversableLike.map(TraversableLike.scala:233)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:226)
        at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:194)
        at com.github.fsanaulla.chronicler.core.model.QueryResult.map(InfluxResult.scala:79)
        at com.github.fsanaulla.chronicler.urlhttp.io.api.Measurement.$anonfun$read$1(Measurement.scala:74)
        at scala.util.Success.$anonfun$map$1(Try.scala:251)
        at scala.util.Success.map(Try.scala:209)
        at com.github.fsanaulla.chronicler.urlhttp.io.api.Measurement.read(Measurement.scala:73)
        at com.github.fsanaulla.chronicler.examples.read.DomainRead$.delayedEndpoint$com$github$fsanaulla$chronicler$examples$read$DomainRead$1(DomainRead.scala:75)
        at com.github.fsanaulla.chronicler.examples.read.DomainRead$delayedInit$body.apply(DomainRead.scala:27)
        at scala.Function0.apply$mcV$sp(Function0.scala:34)
        at scala.Function0.apply$mcV$sp$(Function0.scala:34)
        at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
        at scala.App.$anonfun$main$1$adapted(App.scala:76)
        at scala.collection.immutable.List.foreach(List.scala:388)
        at scala.App.main(App.scala:76)
        at scala.App.main$(App.scala:74)
        at com.github.fsanaulla.chronicler.examples.read.DomainRead$.main(DomainRead.scala:27)
        at com.github.fsanaulla.chronicler.examples.read.DomainRead.main(DomainRead.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sbt.Run.invokeMain(Run.scala:93)
        at sbt.Run.run0(Run.scala:87)
        at sbt.Run.execute$1(Run.scala:65)
        at sbt.Run.$anonfun$run$4(Run.scala:77)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
        at sbt.util.InterfaceUtil$$anon$1.get(InterfaceUtil.scala:10)
        at sbt.TrapExit$App.run(TrapExit.scala:252)
        at java.lang.Thread.run(Thread.java:748)

DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)
DomainEntity(a,b,1,2.3,11)

expected string, got number
[success] Total time: 4 s, completed 04.11.2018. 21:04:20
```

**Comment by fsanaulla** [comment]
When you pass `epoch` param to your query, the result will be returned as `Long` otherwise it will return UTC string.

**Comment by fsanaulla** [comment]
Solved in #105.
